keps/sig-node/1985-remove-dockershim/README.md

```diff
@@ -0,0 +1,302 @@
+- The number of affected users may be large.
+  - Users must change existing use experience when using Kubernetes and docker.
+  - Users have to change their existing workflows to adapt to this new changes.
+  - And other unrecorded stuff.
+- CRI is still technically in alpha, and should graduate to GA before removing dockershim from kubelet. There 
+is a [KEP 2041](https://github.com/kubernetes/enhancements/pull/2041) for graduating the CRI API version.
+- cri-dockerd will vendor kubernetes/kubernetes, that may be tough.
+- cri-dockerd as an independent software running on node should be allocated enough resource to guarantee its availability.
+
+> You can check [the discussion in sig-node mailing list](https://groups.google.com/forum/#!msg/kubernetes-sig-node/0qVzfugYhro/l6Au216XAgAJ) for more details. 
+
+### Goals
+
+- A concrete dockershim removal criteria.
+- A brief plan to remove dockershim spanning multiple releases.
+
+### Non-Goals
+
+- Refactoring or re-design of dockershim itself due to deprecation.
+
+## Proposal
+
+### Dockershim removal criteria
+
+- CRI itself is alpha. So we need another KEP to graduate CRI API.
+- kubelet has no dependency on dockershim/docker in its whole lifecycle. This is already done using the `dockerless` tag
+- All node related features are CRI generic and have no "back door" dependency on dockershim/docker.
+- Deprecate and remove, or replace all Docker-specific features.
+- Reasonable benchmark result of performance degradation after moving dockershim to out-of-tree.
+- E2E test framework has been updated with fully support of out-of-tree CRI container runtime.
+
+### Dockershim removal plan
+
+Step 1: Deprecate in-tree dockershim and decouple dockershim from kubelet.
+
+Target releases: 1.20, 1.21
+
+Actions:
+
+- Mark in-tree dockershim as "maintenance mode":
+  - CRI generic changes/features can continue on dockershim.
+  - WIP efforts on dockershim can continue and go to complete.
+  - dockershim/docker specific changes/features should be rejected.
+- Deprecate the legacy features of dockershim in kubelet by providing a specific timeline. Currently, kubelet still has:
+  - vendored dockershim 
+  - flags that are used to configure dockershim.
+  - support to get container logs when docker uses journald as the driver.
+  - logic of moving docker processes to a given cgroup
+- Ensure e2e/Node e2e test framework is CRI generic and test cases are independent of container runtime.
+- Refactoring e2e/Node e2e test framework to include CRI for docker installation (or use other CRI container runtime).
+  - Ensure cluster/node e2e are 100% CRI focused.
+  - Ensure test-infra install appropriate CRI implementations in e2e machines. 
+- Ensure Windows scenarios that currently depend on docker are fully supported in alternative CRI implementations
+
+Step 2: Release kubelet without dockershim
+
+Target releases: 1.22
+
+Actions:
+- Document and announce migration guide.
```

Comment:
is there a tracking issue for this migration guide, i think it would help to include in it what steps do existing users take for moving from docker to containerd running on there hosts ?

Reply:
I started this: https://github.com/kubernetes/website/pull/25787